core/main/src/bootstrap/extn/load_session_step.rs

```diff
@@ -30,16 +46,149 @@ impl Bootstep<BootstrapState> for LoadDistributorValuesStep {
     }
 
     async fn setup(&self, s: BootstrapState) -> RippleResponse {
-        MainContextProcessor::remove_expired_and_inactive_entries(&s.platform_state);
+        remove_expired_and_inactive_entries(&s.platform_state);
 
         if !s.platform_state.supports_session() {
             return Ok(());
         }
-
-        s.platform_state
-            .get_client()
-            .add_event_processor(MainContextProcessor::new(s.platform_state.clone()));
+        setup_ripple_context_event_handler(s.platform_state);
 
         Ok(())
     }
 }
+
+fn setup_ripple_context_event_handler(state: PlatformState) {
+    let (tx, mut rx) = mpsc::channel::<ServiceMessage>(1);
```

Comment:
What is the reason for setting the channel size to 1 ? What happens when there are more than 1 events received? Are the sender using `try_send` or `send` function (blocked call)